src/app/filters/filter-fields.component.html

```diff
@@ -3,7 +3,7 @@
     <div class="input-group-btn" dropdown>
```

Comment:
With these mods, setupConfig is skipped when fields.length = 0. In that case it may be better to just hide the filter? I doesn't make a lot of sense to show a filter with an empty title, no fields, no options, etc.

Instead of adding a ? for every property, I would prefer to modify the setupConfig method to use default config values. For example, below is the default config and setup from the listView component.

defaultConfig = {
    selectItems: false,
    multiSelect: false,
    dblClick: false,
    dragEnabled: false,
    selectedItems: [],
    selectionMatchProp: 'uuid',
    checkDisabled: false,
    useExpandingRows: false,
    showSelectBox: true
  } as ListViewConfig;

  constructor() {
  }

  // Initialization

  ngOnInit(): void {
    this.setupConfig();
  }

  ngDoCheck(): void {
    // Do a deep compare on config
    if (!_.isEqual(this.config, this.prevConfig)) {
      this.setupConfig();
    }
    this.itemsEmpty = !(this.items !== undefined && this.items.length > 0);
  }

  setupConfig(): void {
    if (this.config !== undefined) {
      _.defaults(this.config, this.defaultConfig);
    } else {
      this.config = _.cloneDeep(this.defaultConfig);
    }
    if (this.config && (this.config.multiSelect === undefined || this.config.multiSelect === false)
        && this.config.selectedItems && this.config.selectedItems.length > 0) {
      this.config.selectedItems = [this.config.selectedItems[0]];
    }
    if (this.config && this.config.selectItems && this.config.showSelectBox) {
      throw new Error('ListViewComponent - Illegal use: ' +
        'Cannot use both select box and click selection at the same time.');
    }
    this.prevConfig = _.cloneDeep(this.config);
  }